### ISSUE TITLE
mongodbatlas_network_container Already exists

### ISSUE THREAD

**Issue by nitrag** [mongodb/terraform-provider-mongodbatlas#88]
I think this is the valid scenario to reproduce the issue:

Terraform:
1.  In Terraform, Create project, container, cluster, VPC peering to AWS VPC, from scratch.
2.  Outside of Terraform, in dashboard, add a new cluster to that project.
3. In Terraform, destroy resources. Successful. Project remains though because manually created cluster.
4. In Terraform, try and recreate cluster (just destroyed) with same variables. You will get error `A container already exists for group`.  (Also get similar project error but it makes sense to do `terraform import` for that)

When querying the API manually to GET containers, it is returning `provisioned` to a AWS VPC that no longer exists (we destroyed the VPC peering with the initial cluster):
```
    "results": [
        {
            "atlasCidrBlock": "10.251.0.0/24",
            "id": "5d4eeb35cf09a24bd2acec99",
            "providerName": "AWS",
            "provisioned": true,
            "regionName": "US_EAST_1",
            "vpcId": "vpc-02b97143f09238a26"
        }
    ],
```

**Expected**: 
Ignore existing container with old VPC and create new one. Maybe Atlas backend hasn't purged it yet?

**Workaround**:
 Import existing container so terraform can apply/deploy the new cluster.

**BUG after deploy:** 

- GET https://cloud.mongodb.com/api/atlas/v1.0/groups/my-project/containers/
 Is still showing the (original) single container with the wrong VPC ID. 
- In the dashboard, the correct VPC peering is displayed and showing healthy.

**Comment by themantissa** [comment]
@nitrag 

I couldn't quite reproduce this but this what I did and what I found:

1)  In Terraform, Create project, container, cluster, VPC peering to AWS VPC, from scratch in US_EAST_1
2) Post create, Checked that all created successfully in the UI and via API as well and that a container was created - noted that  "provisioned" was set to true (this is important for later)
2) Outside of Terraform, in dashboard, add a new cluster to that project.  (same region, size, etc as terraform created cluster)
3) In Terraform, destroy resources. Was not successful - exits on trying to destroy the container with the error message:
`Error: error deleting MongoDB Network Peering Container (REDACTED): couldn't find resource (21 retries)`
Project also not deleted.
4) Listed containers for the project, still there and noted that provisioned is still set to true.
5) Delete cluster created outside of Terraform.
6) Listed containers for the project, noted that provisioned is now false
7) In Terraform destroy resources, now container and project are deleted.

So as long as a cluster exists in a project that is in the same region as the container the container's provisioned value is set to true.  Since one can only use the DELETE endpoint for containers when it's unprovisioned we can't remove it until all the clusters in the same region are removed.  Terraform could remove it once I deleted the non-terraform created cluster.

I also tested this by creating the the terraform cluster in US_EAST_1 and then the UI cluster in US__WEST_1.  This creates 2 containers since in AWS we need a container per AWS region.  This time the destroy worked for all except the project.  A project can't be deleted with a cluster still in it.  Once I deleted the non-terraform cluster I could use terraform to delete the project.

Hence it seems unlikely that the container and project in step 3 were actually deleted, it likely (if in the same region as the non-terraform cluster) should have exited on the container destruction.  That's seems to be the likeliest reason you still see the same container.

**Comment by themantissa** [comment]
Closing since situation was triggered by manual work in a Terraform managed environment and repo shows all is working as intended.

**Comment by nitrag** [comment]
Thanks for looking into this and I appreciate your explanation.

**Comment by leynebe** [comment]
Hi

Just want to share my experience with this error as well. 
I run infrastructure E2E tests which creates and destroys our entire infra, this involves spinning up a MongoDBAtlas instance and destroying it again after we successfully tested our infra deployment works with it.
I've gotten this error on and off over the last months. I initially thought our upgrade to the MongoDBAtlas terraform provider `1.1.0` solved it but then saw it again and noticed a similar ticket being closed without solution on the `1.2.0-pre.1` [release notes](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md#v120-pre1-2022-01-13) which prompted me to make this comment.

The only resource left in our entire stack is the networking container:
![image](https://user-images.githubusercontent.com/34703148/154644894-95c2c39f-5e51-4177-bf2b-350cff5bc6bb.png)
![image](https://user-images.githubusercontent.com/34703148/154644437-27385d9b-cb2c-4e59-86ba-4745720f1f65.png)
 
When executing `mongocli atlas networking containers list` I indeed notice the network container `PROVISIONED` parameter is set to `true`. This would imply I still have some databases/clusters running in the project if I understand correctly. However, each project has it's own Terraform codebase and no databases are ever deployed manually. I've reviewed the project via the web interface and I'm not seeing a database, no data lake, no triggers, no data api, no database users, no network access peering or private endpoints. Are there some obscure resources I'm not mentioning which still cause the `PROVISIONED` parameter to be set to `true`?

I really appreciate the work that's put in this IAC module, and I'm overall very happy with it, so I don't mind providing additional information to get this resolved. 

Thanks

Ben
